examples/hello-world/format-example.cpp

```diff
@@ -0,0 +1,144 @@
+
+#include "internalGroup1.h"
+#include "slang.h"
+#include <algorithm>
+
+#include "includeGroup2.h"
+#include <vector>
+
+#include <vulkan.h>
+
+#ifdef __linux__
+#    include <X11/Xlib.h>
+#else
+#    include <Windows.h>
+#endif
+
+namespace slang
+{
+/// <summary>
+/// Summary of TestClass
+/// </summary>
+class TestClass
+{
+private:
```

Comment:
The style of having private members at the start of a class is the opposite of what is desired in general. Typically the member variables are an 'implementation detail'.They should typically be at the end of a class definition. 

That the first/main thing you want to see in a class definition is the interface. Under this logic unless the methods implementation is small, it should be placed later, either in a .cpp file or in the header, but afterwards/inlined.

If there are public member variables, they should be before protected:/private:

Reply:
This is not a style that clang-format enforces. It is just that my sample code happen to be this way. Clang-format will not change the order things are declared in classes.